llvm/lib/Analysis/HashRecognize.cpp

```diff
@@ -634,6 +649,19 @@ HashRecognize::recognizeCRC() const {
     return VE.getError();
   KnownBits ResultBits = VE.KnownPhis.at(ConditionalRecurrence.Phi);
 
+  // There must be exactly four unvisited instructions, corresponding to the
+  // IndVar PHI:
+  //   IndVar
+  //   Latch->getTerminator()
+  //   L.getLatchCmpInst(),
+  //   IndVar->getIncomingValueForBlock(Latch))
```

Comment:
Can we instead insert these into visited instead of having a magic "4" in the code and this listing in a comment?